Extend health/progressing checks of `gardener-resource-manager` by `cert-management` resources (#9326)
gardener/gardener: docs/concepts/resource-manager.md

### [`health` Controller](../../pkg/resourcemanager/controller/health)

This controller processes `ManagedResource`s that were reconciled by the main [ManagedResource Controller](#ManagedResource-Controller) once.
Its main job is to perform checks for maintaining the well [known conditions](#conditions) `ResourcesHealthy` and `ResourcesProgressing`.

#### Progressing Checks

In Kubernetes, applied changes must usually be rolled out first, e.g. when changing the base image in a `Deployment`.
Progressing checks detect ongoing roll-outs and report them in the `ResourcesProgressing` condition of the corresponding `ManagedResource`.

The following object kinds are considered for progressing checks:
- `DaemonSet`
- `Deployment`
- `StatefulSet`
- [`Prometheus`](https://github.com/prometheus-operator/prometheus-operator)
- [`Alertmanager`](https://github.com/prometheus-operator/prometheus-operator)

#### Health Checks

> `gardener-resource-manager` can evaluate the health of specific resources, often by consulting the resources' conditions.

Reviewer: ```suggestion
`gardener-resource-manager` can evaluate the health of specific resources, often by consulting the resource's conditions.
```

Author: The plural was meant here, i.e. conditions of resources.

In addition to the origin annotation, all objects managed by the resource manager get a dedicated label `resources.gardener.cloud/managed-by`. This label can be used to describe these objects with a [selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). By default it is set to "gardener", but this can be overwritten by setting the `.conrollers.managedResources.managedByLabelValue` field in the component configuration.

### [`health` Controller](../../pkg/resourcemanager/controller/health)

> This controller processes `ManagedResource`s that were reconciled by the main [ManagedResource Controller](#ManagedResource-Controller) once.

Reviewer 2: Nit:
```suggestion
This controller processes `ManagedResource`s that were reconciled by the main [ManagedResource Controller](#ManagedResource-Controller) at least once.
```

> This controller processes `ManagedResource`s that were reconciled by the main [ManagedResource Controller](#ManagedResource-Controller) at least once.

Reviewer 3: Nit: Is this link working? Aren't those anchors typically lowercase?

Author: It worked but decided to change it for consistency's sake.

- `Deployment`
- `StatefulSet`
- [`Prometheus`](https://github.com/prometheus-operator/prometheus-operator)
- [`Alertmanager`](https://github.com/prometheus-operator/prometheus-operator)
- [`Certificate`](https://github.com/gardener/cert-management)
- [`Issuer`](https://github.com/gardener/cert-management)

#### Health Checks

`gardener-resource-manager` can evaluate the health of specific resources, often by consulting the resources' conditions.
Health check results are regularly updated in the `ResourcesHealthy` condition of the corresponding `ManagedResource`.

The following object kinds are considered for health checks:
- `CustomResourceDefinition`
- `DaemonSet`
- `Deployment`
- `Job`
- `Pod`
- `ReplicaSet`
- `ReplicationController`
- `Service`
- `StatefulSet`
- [`VerticalPodAutoscaler`](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler)
- [`Prometheus`](https://github.com/prometheus-operator/prometheus-operator)
- [`Alertmanager`](https://github.com/prometheus-operator/prometheus-operator)
- [`Certificate`](https://github.com/gardener/cert-management)
- [`Issuer`](https://github.com/gardener/cert-management)

#### Skipping Health Check

> If a resource in the `ManagedResource` is annotated with `resources.gardener.cloud/skip-health-check=true`, then the resource will be skipped during health checks by the `health` controller. The `ManagedResource` conditions will not reflect the health condition of this resource anymore. The `ResourcesProgressing` condition will also be set to `False`.

Reviewer 2: ```suggestion
If a resource owned by a `ManagedResource` is annotated with `resources.gardener.cloud/skip-health-check=true`, then the resource will be skipped during health checks by the `health` controller. The `ManagedResource`'s conditions will not reflect the health condition of this resource anymore. The `ResourcesProgressing` condition will also be set to `False`.
```
"owned by a `ManagedResource`" or "part of a `ManagedResource`" seems a bit better to me here, but no strong opinion.
